app/src/main/java/com/example/cocktaildb/screen/auth/AuthPresenter.kt

```diff
@@ -0,0 +1,179 @@
+package com.example.cocktaildb.screen.auth
+
+import android.app.Activity
+import android.content.Intent
+import com.example.cocktaildb.data.repository.AuthRepository
+import com.example.cocktaildb.data.model.LoginMethod
+import com.example.cocktaildb.utils.GoogleAuth
+
+class AuthPresenter(
+    private val authRepository: AuthRepository
+) : AuthContract.Presenter {
+
+    private var view: AuthContract.View? = null
+    private var googleAuthHelper: GoogleAuth? = null
+
+    override fun setView(view: AuthContract.View?) {
+        this.view = view
+        if (view is Activity) {
+            googleAuthHelper = GoogleAuth(view as Activity)
+        }
+    }
+
+    override fun onStart() {
+        // TODO: Initialize if needed
+    }
+
+    override fun onStop() {
+        // TODO: Cleanup if needed
+    }
+
+    override fun loginWithEmail(email: String, password: String) {
+        if (!validateLoginInput(email, password)) return
+
+        view?.showLoading()
+        authRepository.loginWithEmail(email, password) { firebaseUser, error ->
+            view?.hideLoading()
+            if (firebaseUser != null) {
+                view?.onLoginSuccess()
+            } else {
+                view?.onLoginFailure(error ?: "Email hoặc password không đúng")
+            }
+        }
+    }
+
+    override fun signUpWithEmail(
+        name: String,
+        email: String,
+        password: String,
+        confirmPassword: String,
+        termsAccepted: Boolean
+    ) {
+        if (!validateSignUpInput(name, email, password, confirmPassword, termsAccepted)) return
+
+        view?.showLoading()
+        authRepository.signUpWithEmail(name, email, password) { firebaseUser, error ->
+            view?.hideLoading()
+            if (firebaseUser != null) {
+                view?.onLoginSuccess()
+            } else {
+                view?.onLoginFailure(error ?: "Đăng ký thất bại")
+            }
+        }
+    }
+
+    override fun loginWithGoogle() {
+        googleAuthHelper?.let { helper ->
+            view?.showLoading()
+            val signInIntent = helper.getSignInIntent()
+            if (view is SignInActivity) {
+                (view as SignInActivity).launchGoogleSignIn(signInIntent)
+            } else if (view is SignUpActivity) {
+                (view as SignUpActivity).launchGoogleSignIn(signInIntent)
+            } else {
+                view?.showMessage("Google login không khả dụng")
```

Comment:
```suggestion
                view?.showMessage(R.string.google_sign_in_not_available)
```
Không hard code và default sử dụng tiếng việt trong code, sử dụng tiếng anh cho nó quốc tế.

- Khai báo ``google_sign_in_not_available`` trong folder ``res/values/strings.xml`` thêm tiếng việt thì khai báo trong folder ``res/values-vi/strings.xml``

sử tương tự với những chỗ khác